app/code/Magento/QuoteGraphQl/Model/Resolver/UpdateCartItems.php

```diff
@@ -94,53 +111,47 @@ private function processCartItems(Quote $cart, array $items): void
             if (!isset($item['cart_item_id']) || empty($item['cart_item_id'])) {
                 throw new GraphQlInputException(__('Required parameter "cart_item_id" for "cart_items" is missing.'));
             }
-            $itemId = $item['cart_item_id'];
+            $itemId = (int)$item['cart_item_id'];
 
             if (!isset($item['quantity'])) {
                 throw new GraphQlInputException(__('Required parameter "quantity" for "cart_items" is missing.'));
             }
             $quantity = (float)$item['quantity'];
 
-            $cartItem = $cart->getItemById($itemId);
-            if ($cartItem === false) {
-                throw new GraphQlNoSuchEntityException(
-                    __('Could not find cart item with id: %1.', $item['cart_item_id'])
-                );
-            }
-
             if ($quantity <= 0.0) {
                 $this->cartItemRepository->deleteById((int)$cart->getId(), $itemId);
             } else {
-                $cartItem->setQty($quantity);
-                $this->validateCartItem($cartItem);
-                $this->cartItemRepository->save($cartItem);
+                $customizableOptions = $item['customizable_options'] ?? null;
+
+                if ($customizableOptions === null) { // Update only item's qty
+                    $this->updateItemQty($itemId, $cart, $quantity);
+                } else { // Update customizable options (and QTY if changed)
+                    $this->updateCartItem->execute($cart, $itemId, $quantity, $customizableOptions);
+                    $this->quoteRepository->save($cart);
+                }
             }
         }
     }
 
     /**
-     * Validate cart item
+     * Updates item qty for the specified cart
      *
-     * @param Item $cartItem
-     * @return void
-     * @throws GraphQlInputException
+     * @param int $itemId
+     * @param Quote $cart
+     * @param float $qty
+     * @throws GraphQlNoSuchEntityException
+     * @throws NoSuchEntityException
+     * @throws GraphQlNoSuchEntityException
      */
-    private function validateCartItem(Item $cartItem): void
```

Comment:
Please note that all the validations and error handling are moved here https://github.com/magento/graphql-ce/pull/626/files#diff-23947b0596c6c269244b8667cb99db1aR46